shield: m5stack_m5dial: Add support for m5dial shield (#67253)
zephyrproject-rtos/zephyr: boards/shields/m5stack_m5dial/doc/index.rst

.. _m5stack_m5dial:

M5Stack M5Dial
##############

Overview
********

M5Stack M5Dial is an ESP32-based development board from M5Stack.
The device features an embedded :ref:`m5stack_stamps3` module that runs
the ESP32-S3 microcontroller.

> .. figure:: img/m5stack_m5dial.webp
>         :align: center
>         :alt: M5Stack M5Dial
> 
>         M5Stack M5Dial

Reviewer: ```suggestion
.. figure:: img/m5stack_m5dial.webp
   :align: center
   :alt: M5Stack M5Dial

        M5Stack M5Dial
```

.. _m5stack_m5dial:

M5Stack M5Dial
##############

Overview
********

M5Stack M5Dial is an ESP32-based development board from M5Stack.

> The device features an embedded M5Stack-StampS3 module that runs

Reviewer: :) needs the global search & replace in the context of M5Dial too, I guess!
Also, it might make sense to make all mentions (or at least this one as this if the first one in the page) to StampS3 a link to the board page

```suggestion
The device features an embedded M5Stack StampS3 module that runs
```

| M1-26    |                                |         | TX        |                                   |
+----------+--------------------------------+---------+-----------+-----------------------------------+
| M1-27    | ``m5stack_stamps3_header:26``  |         | HOLD      | Battery-hold signal               |
+----------+--------------------------------+---------+-----------+-----------------------------------+
|| M1-28   || **3_3V**                                            || 3.3V main output                 |
||         ||                                                     || (via StampS3 DCDC)               |
+----------+--------------------------------+---------+-----------+-----------------------------------+

Start Application Development
*****************************

Before powering up your M5Stack M5Dial, please make sure that the board is in good
condition with no obvious signs of damage.

System requirements
===================

Prerequisites
-------------

Espressif HAL requires WiFi and Bluetooth binary blobs in order work. Run the command
below to retrieve those files.

.. code-block:: shell

   west blobs fetch hal_espressif

.. note::

>    It is recommended running the command above after :file:`west update`.

Reviewer: remove

Building & Flashing
-------------------

M5Dial is based on :ref:`m5stack_stamps3`.

> Build and flash applications as usual (see :ref:`build_an_application` and
> :ref:`application_run` for more details).
> 
> A good first example to test is :ref:`hello_world_lvgl`. The usual ``flash``
> target will work with the ``m5stack_stamps3`` board and ``m5stack_m5dial``
> shield configuration:
> 
> .. zephyr-app-commands::
>    :zephyr-app: samples/subsys/display/lvgl
>    :board: m5stack_stamps3
>    :shield: m5stack_m5dial
>    :goals: flash

Reviewer: ```suggestion
You may build and flash applications for the M5Stack StampS3 as usual (see :ref:`build_an_application` and
:ref:`application_run` for more details), and include this shield as an overlay.

A good first example to test is :zephyr:code-sample:`hello_world_lvgl`:

.. zephyr-app-commands::
   :zephyr-app: samples/subsys/display/lvgl
   :board: m5stack_stamps3
   :shield: m5stack_m5dial
   :goals: build flash
```

The baud rate of 921600bps is set by default. If experiencing issues when flashing,
try using different values by using ``--esp-baud-rate <BAUD>`` option during
``west flash`` (e.g. ``west flash --esp-baud-rate 115200``).

After the board has automatically reset and booted, you should see the following
content on the screen:

.. figure:: img/helloworld_screenshoot.webp
        :align: center
        :alt: M5Stack M5Dial

>         :width: 200px

Reviewer: I would remove this (and the screenshot -- not screenshoot ;-) file too) since if all boards/shields start adding text or screen captures showing how the output of a particular sample is supposed to look, it comes with a very good chance that these will become stale very quickly if said output changes overtime. And it's not really adding a lot of value to the end user anyway.
